### PR TITLE
[Homework #18] - Hooks by Abraham Trenyan

### PR DESCRIPTION
**Pull request by AbrahamTrenyan**

### CODE REVIEW

File: homeworks/abraham-trenyan/18-hooks/src/components/Faces/Faces.jsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import './Faces.css';
+import { useState } from 'react';
+import RefreshIcon from '../RefreshIcon/RefreshIcon';
+const Faces = () => {
+  const [images, setImages] = useState([]);
+  const addImg = async () => {
+    let response = await fetch(`https://tinyfac.es/api/data?limit=1&quality=0`);
+    response.json().then((result) => {
```
Comment (VitalikMelnyk) [inline homeworks/abraham-trenyan/18-hooks/src/components/Faces/Faces.jsx:8]: Try to follow my notes from previous homework about avoiding mixing `async/await` with `try/catch`

**Review by VitalikMelnyk — APPROVED** [review]

**Comment by VitalikMelnyk** [comment]
Hello @AbrahamTrenyan 
Review my note!
Thanks.

**Comment by AbrahamTrenyan** [comment]
> Hello @AbrahamTrenyan Review my note! Thanks.

Sorry Vitalik, i was working on on another homework! Now its done

File: homeworks/abraham-trenyan/18-hooks/src/components/Faces/Faces.jsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import './Faces.css';
+import { useState } from 'react';
+import RefreshIcon from '../RefreshIcon/RefreshIcon';
+const Faces = () => {
+  const [images, setImages] = useState([]);
+  const addImg = async () => {
+    const response = await fetch(`https://tinyfac.es/api/data?limit=1&quality=0`);
```
Comment (VitalikMelnyk) [inline homeworks/abraham-trenyan/18-hooks/src/components/Faces/Faces.jsx:7]: By the way, I noticed repeatable code.
Please, create a shared function for getting data from API.

File: homeworks/abraham-trenyan/18-hooks/src/components/Faces/Faces.jsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    const result = await response.json();
+    setImages([...images, result[0].url]);
+  };
+  const refreshAll = async () => {
+    const response = await fetch(`https://tinyfac.es/api/data?limit=${images.length}&quality=0`);
```
Comment (VitalikMelnyk) [inline homeworks/abraham-trenyan/18-hooks/src/components/Faces/Faces.jsx:12]: Here.

File: homeworks/abraham-trenyan/18-hooks/src/components/Faces/Faces.jsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    const result = await response.json();
+    let newImgs = [];
+    for (let i = 0; i < result.length; i++) {
+      newImgs.push(result[i].url);
+    }
+    setImages([...newImgs]);
+  };
+  const refreshImg = async (e) => {
+    const response = await fetch(`https://tinyfac.es/api/data?limit=1&quality=0`);
```
Comment (VitalikMelnyk) [inline homeworks/abraham-trenyan/18-hooks/src/components/Faces/Faces.jsx:21]: Here.

Comment (AbrahamTrenyan) [reply homeworks/abraham-trenyan/18-hooks/src/components/Faces/Faces.jsx]: now  its all in one function!

**Review by VitalikMelnyk — APPROVED** [review]

**Pull request merged**